src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -357,33 +410,61 @@ std::string Bmi_Multi_Formulation::get_output_line_for_timestep(int timestep, st
+        //   - tried a more complex config, but ran into an error, and are needing to revert to the default
+        return modules.back()->get_output_line_for_timestep(timestep, delimiter);
+    }
+    else{    
         // TODO: see Github issue 355: this design (and formulation output handling in general) needs to be reworked
         // Clear anything currently in the multi formulation's stream buffer
         output_text_stream->str(std::string());
 
         const std::vector<std::string> &output_var_names = get_output_variable_names();
+        const std::vector<std::string> &output_var_units = get_output_variable_units();
+
         // This almost certainly should never happen, but just to be safe ...
         if (output_var_names.empty()) { return ""; }
 
-        // Do the first separately, without the leading comma
-        *output_text_stream << get_var_value_as_double(0, output_var_names[0]);
-
-        // Do the rest with a leading comma
-        for (int i = 1; i < output_var_names.size(); ++i) {
-            *output_text_stream << delimiter << get_var_value_as_double(0, output_var_names[i]);
+        for (int i = 0; i < output_var_names.size(); ++i) {
+            double value;
+            try{
+                value = get_value(CatchmentAggrDataSelector(get_catchment_id(), output_var_names[i], 0, 0, output_var_units[i]), MEAN);
+            }
+            catch(data_access::unit_conversion_exception &uce){
+                data_access::unit_error_log_key key{"File Output Multi", output_var_names[i], uce.provider_model_name, uce.provider_bmi_var_name, uce.what()};
+                auto ret = data_access::unit_errors_reported.insert(key);
+                bool new_error = ret.second;
+                if (new_error) {
+                    std::stringstream ss;
+                    ss << "Unit conversion failure:"
+                        << " requester {'Get Output Line for Timestep (Multi Formulation)"
+                        << "' catchment '" << get_catchment_id()
+                        << "' variable '" << output_var_names[i]
+                        << "' units '" << output_var_units[i] << "'}" 
+                        << " provider {'" << uce.provider_model_name 
+                        << "' source variable '" << uce.provider_bmi_var_name << "'"
+                        << " raw value " << uce.unconverted_values[0] << "}"
+                        << " message \"" << uce.what() << "\"";
+                    LOG(ss.str(), LogLevel::WARNING); ss.str("");
+                }
+                value = uce.unconverted_values[0];
+            }
+            std::string var_value = std::to_string(value);
+            if(i == 0){
+                *output_text_stream << var_value; //without delimiter for first output variable.
+            }
+            else{
+                *output_text_stream << delimiter << var_value; //with delimiter for the rest.
```

Comment:
To not have to get into fixing some over-sensitive tests, we should allow `output_text_stream` to do the formatting, so that it can exactly match the expected precision, width, etc:
```suggestion
            if(i == 0){
                *output_text_stream << value; //without delimiter for first output variable.
            }
            else{
                *output_text_stream << delimiter << value; //with delimiter for the rest.
```